RPIPs/RPIP-31.md

```diff
@@ -28,18 +28,45 @@ https://dao.rocketpool.net/t/withdrawal-address-splitter-contract/563
 
 ## Specification
 - The withdrawal address that has existed to date SHALL be known as the primary withdrawal address
+
+### Defaults
+- The RPL withdrawal address SHALL be initialized as unset for new node operators
+- The RPL withdrawal address SHALL be initialized as unset for existing node operators
+  
+### Rewards & Withdrawn RPL
 - If a node's RPL withdrawal address is set, all RPL rewards and withdrawn RPL will go to the RPL withdrawal address
 - If a node's RPL withdrawal address is unset, all RPL rewards and withdrawn RPL will go to the primary withdrawal address
+
+### Setting the Address
 - As a node operator, I MUST be able to set an RPL withdrawal address so that I can direct my RPL bond/rewards
   - If a node's RPL withdrawal address is unset, the change MUST come from the node's primary withdrawal address
   - If a node's RPL withdrawal address is set, the change MUST come from the current RPL withdrawal address
 - As a node operator, I MUST be able to confirm the RPL withdrawal address from the new target address before it takes effect, to ensure it is the correct address
 - As the owner of the RPL withdrawal address, I MUST be able to unset my RPL withdrawal address
+
+### Withdrawing Excess RPL
 - As the controller of the RPL withdrawn from a node, I MUST be able to call `RocketNodeStaking.withdrawRPL()` to withdraw excess RPL
   - If a node's RPL withdrawal address is unset, the call MUST come from one of: the node's primary withdrawal address, or the node's address
   - If a node's RPL withdrawal address is set, the call MUST come from the current RPL withdrawal address
-- The RPL withdrawal address SHALL be initialized as unset for new node operators
-- The RPL withdrawal address SHALL be initialized as unset for existing node operators
+
+### Claiming RPL rewards
+- As the controller of the RPL for a node, I MUST be able to call `RocketMerkleDistributorMainnet.claim()` to trigger a claim of RPL rewards
+  - If a node's RPL withdrawal address is unset, the call MUST come from one of: the node's primary withdrawal address, or the node's address
+  - If a node's RPL withdrawal address is set, the call MUST come from one of: the node's primary withdrawal address, the current RPL withdrawal address, or the node's address
```

Comment:
🤔 not sure I get this. If the premise is that we have up to 3 totally separate parties - ETH investor, RPL investor, Node Operator - why should all 3 be able to decide when to send RPL rewards out?

I can make a case for the NO, as this is the most "turnkey" approach where the investors need not act at all, but I don't see a case for the ETH investor to have this ability.

Reply:
For this, I have matched what is currently implemented: **withdrawal address AND node address** and added the RPL withdrawal address.

This is so that a node operator cannot withhold rewards and that is it operationally convenient. This would be the same for an ETH investor as well.

Reply:
Not tracking - why can the ETH investor force the RPL investor to claim (and not vice versa)?

Reply:
I feel I am missing something as well lol. 

When the RPL address is set, both the ETH investor and RPL investor can trigger a reward claim. I don't dig the fact that smoothing pool rewards and RPL rewards are coupled together but implementation-wise it is much more complicated to separate them.

Reply:
Got it, it's cuz of the smoothing pool. If not in smoothing pool does that still apply?

Reply:
Currently, the node address and primary withdrawal address can call `RocketMerkleDistributorMainnet.claim()` regardless of whether they are in the smoothing pool or not.

We could revert node and primary withdrawal address calls if they are not in the smoothing pool but it then creates an inconsistent experience for regular users that use the RPL withdrawal address - as soon as they join the smoothing pool they can no longer claim from their node or primary withdrawal address.